### ISSUE TITLE
Application/json validator is the geojson validator

### ISSUE THREAD

**Issue by huard** [geopython/pywps#390]
# Description
The complexvalidator for json type files is the geojson validator, whichl raises an error for valid json files. 

# Environment
Master on Ubuntu 16.04. 

# Steps to Reproduce
There is yet no test to exercise this issue. 

# Additional Information

**Comment by huard** [comment]
I'm not sure what the SIMPLE validation actually does: 
```
passed = data_input.data_format.mime_type in {mtype, FORMATS.JSON.mime_type}
```
In which situations would it actually catch an error not already checked by the BasicComplex.data_format setter ? 

Shouldn't it compare mtype with data_input.data_format.mime_type ?

**Comment by cehbrecht** [comment]
I was also wandering about the simple validator. My guess was that it should check if the detected mimetype of the input is in the allowed mimetypes.

@jachym Can you put some light on this :)

**Comment by cehbrecht** [comment]
fixed by PR #403.

"simple validator" discussion moved to #407.
